ovirt/resource_ovirt_vm.go

```diff
@@ -613,32 +613,23 @@ func (c *providerContext) resourceOvirtVMCreate(d *schema.ResourceData, meta int
 	if _, ok := d.GetOk("auto_pinning_policy"); ok && versionCompareResult < 0 {
 		log.Printf("[WARN] The engine version %d.%d.%d is not supporting the auto pinning feature. "+
 			"Please update to 4.4.5 or later.", engineVer.MustMajor(), engineVer.MustMinor(), engineVer.MustBuild())
-	} else {
-		// Mimic the UI behavior. unless specified set to existing policy for high performance VMs.
-		if _, ok := d.GetOk("auto_pinning_policy"); !ok && isHighPerformance {
-			err := d.Set("auto_pinning_policy", ovirtsdk4.AUTOPINNINGPOLICY_EXISTING)
+	} else if v, ok := d.GetOk("auto_pinning_policy"); ok {
```

Comment:
I think this if should be split into two, the second being the versionCompareResult. That would make the code much more readable.

Reply:
I updated the code, let me know WDYT